server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -305,10 +309,73 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failDecommissionedNodes(ClusterState state) {
+    private void drainNodesWithDecommissionedAttribute(ClusterState state, DecommissionRequest decommissionRequest) {
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.DRAINING, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+                checkWeightsAndScheduleDecommission(decommissionRequest);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to update decommission status for attribute [{}] to [{}]",
+                        decommissionRequest.getDecommissionAttribute().toString(),
+                        DecommissionStatus.DRAINING
+                    ),
+                    e
+                );
+                // since we are not able to update the status, we will clear the voting config exclusion we have set earlier
+                clearVotingConfigExclusionAndUpdateStatus(false, false);
+            }
+        });
+    }
+
+    void checkWeightsAndScheduleDecommission(DecommissionRequest decommissionRequest) {
+        ClusterState state = clusterService.getClusterApplierService().state();
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
+        if (decommissionAttributeMetadata == null) {
+            clearVotingConfigExclusionAndUpdateStatus(false, false);
+            return;
+        }
+        assert decommissionAttributeMetadata.status().equals(DecommissionStatus.DRAINING)
+            : "Unexpected status encountered while decommissioning nodes.";
+        DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
```

Comment:
When after the `DRAINING` begins can metadata be null?

Reply:
I think yes.. Delete decommission API can be called which can clear the metadata